pycode/memilio-epidata/memilio/epidata/getNPIData.py

```diff
@@ -895,39 +942,39 @@ def get_npi_data(fine_resolution=2,
             df_infec_local = df_infec_local[(df_infec_local[dd.EngEng['date']] >= start_date_new) & (
                 df_infec_local[dd.EngEng['date']] <= end_date_new)].reset_index()
 
-            local_incid = df_infec_local['Incidence'].copy()
+            local_incid = copy.deepcopy(df_infec_local['Incidence'])
             # Count counties with start cases >= 1:
             # In this case NPI activation cannot be ensured to work as expected
             if cases_first_value >= 1:
                 counter_cases_start += 1
 
         # get county-local data frame
         start_time = time.perf_counter()
-        df_local_old = df_npis_old[df_npis_old[dd.EngEng['idCounty']]
-                                   == countyID].copy()
+        df_local_old = copy.deepcopy(df_npis_old[df_npis_old[dd.EngEng['idCounty']]
+                                                 == countyID])
 
         # Consistency of incidence dependent NPIs:
         # The same NPI should not be prescribed multiple times at the same day
         # for different thresholds. In order to avoid contradictions, only
         # retain the strictest mentioned implementation.
         for i in range(int(len(df_local_old)/6)):
             sum_npi_inc = np.where(
-                df_local_old.iloc[6*i+1:6*(i+1), 6:].sum() > 1)
+                df_local_old.iloc[6*i:6*(i+1), 6:].sum() > 1)
```

Comment:
we should think of replacing the two 6's by different variables. The first represents the number of lines per NPI, the second the start of the NPI columns. Also applies to other occasions.

Reply:
Defined 2 new variables